File: practice/convert/example2_book_library/java/book-library/src/main/java/dev/starryeye/book_library/application/user/query/GetUsersService.java
package dev.starryeye.book_library.application.user.query;

import dev.starryeye.book_library.application.user.query.result.GetUsersResult;
import dev.starryeye.book_library.domain.user.User;
import dev.starryeye.book_library.domain.user.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Transactional(readOnly = true)
@Service
@RequiredArgsConstructor
public class GetUsersService {

    private final UserRepository userRepository;

    public GetUsersResult get() {

        List<User> users = userRepository.findAll();

        return GetUsersResult.of(users);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/user/query/GetUsersService.kt
package dev.starryeye.book_library.application.user.query

import dev.starryeye.book_library.application.user.query.result.GetUsersResult
import dev.starryeye.book_library.domain.user.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional(readOnly = true)
@Service
class GetUsersService(
    private val userRepository: UserRepository,
) {

    fun getAll(): GetUsersResult {
        val users = userRepository.findAll()

        return GetUsersResult.of(users)
    }
}